### PR TITLE
fix: remove the open classes if no visible reveal exists after destro…

### PR DESCRIPTION
**Pull request by DanielRuf**
…ying one

Closes https://github.com/zurb/foundation-sites/issues/10626

### CODE REVIEW

**Review by ncoden — APPROVED** [review]
~~LGTM 👍~~

File: js/foundation.reveal.js [hunk 1]
```diff
@@ -409,7 +414,7 @@ class Reveal extends Plugin {
       var scrollTop = parseInt($("html").css("top"));
 
       if ($('.reveal:visible').length  === 0) {
-        $('html').removeClass('is-reveal-open');
+        removeRevealOpenClasses();
```
Comment (ncoden) [inline js/foundation.reveal.js]: Please use `this.removeRevealOpenClasses();` (and run the docs and tests before creating a PR)

File: js/foundation.reveal.js [hunk 2]
```diff
@@ -469,6 +474,10 @@ class Reveal extends Plugin {
     this.$element.hide().off();
     this.$anchor.off('.zf');
     $(window).off(`.zf.reveal:${this.id}`);
+
+    if ($('.reveal:visible').length  === 0) {
```
Comment (ncoden) [inline js/foundation.reveal.js]: A comment would be great. Also, do you think the test should be duplicated or moved to `removeRevealOpenClasses` ?

File: js/foundation.reveal.js [hunk 3]
```diff
@@ -278,6 +278,11 @@ class Reveal extends Plugin {
       $('html').addClass('is-reveal-open');
     }
 
+    function removeRevealOpenClasses() {
+
```
Comment (ncoden) [inline js/foundation.reveal.js]: empty line

**Review by ncoden — CHANGES_REQUESTED** [review]
Actually...

**Comment by DanielRuf** [comment]
I guess I oversaw a small bit, will change it later.

**Comment by DanielRuf** [comment]
Just forgot the `this.`, will add later =)

File: js/foundation.reveal.js (hunk 3 above)
Comment (DanielRuf) [reply js/foundation.reveal.js]: Will remove it then.

File: js/foundation.reveal.js (hunk 1 above)
Comment (DanielRuf) [reply js/foundation.reveal.js]: Right. Will do so in 7-8 hours, currently not on one of my machines.

File: js/foundation.reveal.js (hunk 2 above)
Comment (DanielRuf) [reply js/foundation.reveal.js]: Not sure, the other function also just adds the class and does no prior check if is is already added.

**Comment by ncoden** [comment]
@DanielRuf Please for each PR:
* run tests
* start the docs
* ensure that no similar change can be made elsewhere
* ensure thet the documentation is up-to-date
* add comments and references, there's never too much comments ;)

**Comment by DanielRuf** [comment]
> run tests
Might be a good idea to create a precommit hook in general. I'll add it to my todo list.

Will do the rest later today.

**Comment by ncoden** [comment]
@DanielRuf Can you share your Todo list (as issues) ?

File: js/foundation.reveal.js (hunk 3 above)
Comment (DanielRuf) [reply js/foundation.reveal.js]: `addRevealOpenClasses` also has an empty line after the function declaration.

File: js/foundation.reveal.js (hunk 2 above)
Comment (ncoden) [reply js/foundation.reveal.js]: If I remember correctly `.addClass()` already do that check.

Comment (DanielRuf) [reply js/foundation.reveal.js]: Meant the check for `:visible`.

Comment (DanielRuf) [reply js/foundation.reveal.js]: Meant the `:visible` check around the calls.

**Comment by DanielRuf** [comment]
We have to check the other failing unit tests on Windows (IE 11 and Firefox 58) as these are different issues.

Who could take a look at them?

File: js/foundation.reveal.js (hunk 2 above)
Comment (ncoden) [reply js/foundation.reveal.js]: We may not have understood each other.

What I'm trying to say is that:
* For now, `addRevealOpenClasses` can be called on all Reveal initializations and will only add the class once (thanks to jQuery `.addClass`). So it takes care of the "check" to put the class only when needed.
* Maybe we should have a similar behavior with `removeRevealOpenClasses`: being callable on all Reveal destructions and take care of the "check" to only remove the class when needed. So with [the check](https://github.com/zurb/foundation-sites/pull/11041/files#diff-5b5e0861de0a3864db2c3c28919700b7L411) inside the function.

But it's a _maybe_, and I don't want to loose too much time on a so small thing 😄

**Review by ncoden — APPROVED** [review]
👍

**Pull request merged**